.github/CONTRIBUTING.md

```diff
@@ -31,141 +31,468 @@ strongly recommended you get approval/traction for it from our forums before sta
+    if(!thing2)
+      if(thing3 == 30)
+        do stuff
+````
+This is good:
+````DM
+/datum/datum1/proc/proc1()
+  if(!thing1)
+    return
+  if(thing2)
+    return
+  if(thing3 != 30)
+    return
+  do stuff
+````
+This prevents nesting levels from getting deeper then they need to be.
+
+### Operators
+#### Spacing
+* Operators that should be separated by spaces
+  * Boolean and logic operators like &&, || <, >, ==, etc (but not !)
+  * Bitwise AND &
+  * Argument separator operators like , (and ; when used in a forloop)
+  * Assignment operators like = or += or the like
+  * Math operators like +, -, /, or \*
+* Operators that should not be separated by spaces
+  * Bitwise OR |
+  * Access operators like . and :
+  * Parentheses ()
+  * logical not !
+
+#### Use
+* Bitwise AND - '&'
+  * Should be written as ```bitfield & bitflag``` NEVER ```bitflag & bitfield```, both are valid, but the latter is confusing and nonstandard.
+* Associated lists declarations must have their key value quoted if it's a string
+  * WRONG: list(a = "b")
+  * RIGHT: list("a" = "b")
+
+### Legacy Code
+SS13 has a lot of legacy code that's never been updated. Here are some examples of common legacy trends which are no longer acceptable:
+  * To display messages to all mobs that can view `src`, you should use
   `visible_message()`.
-     - Bad:
+    * Bad:
      ```
      for(var/mob/M in viewers(src))
              M.show_message("<span class='warning'>Arbitrary text</span>")
      ```
-     - Good:
+    * Good:
      ```
      visible_message("<span class='warning'>Arbitrary text</span>")
      ```
-  - You should not use color macros (`\red, \blue, \green, \black`) to color text,
+  * You should not use color macros (`\red, \blue, \green, \black`) to color text,
   instead, you should use span classes. `<span class='warning'>red text</span>`,
   `<span class='notice'>blue text</span>`.
-    - Bad:
+    * Bad:
     ```
     usr << "\red Red Text \black black text"
     ```
-    - Good:
+    * Good:
     ```
     usr << "<span class='warning'>Red Text</span>black text"
     ```
-  - To use variables in strings, you should **never** use the `text()` operator, use
+  * To use variables in strings, you should **never** use the `text()` operator, use
    embedded expressions directly in the string.
-     - Bad:
+     * Bad:
      ```
      usr << text("\The [] is leaking []!", src.name, src.liquid_type)
      ```
-     - Good:
+     * Good:
      ```
      usr << "\The [src] is leaking [liquid_type]"
      ```
-  - To reference a variable/proc on the src object, you should **not** use
+ * To reference a variable/proc on the src object, you should **not** use
    `src.var`/`src.proc()`. The `src.` in these cases is implied, so you should just use
    `var`/`proc()`.
-     - Bad:
+    * Bad:
      ```
      var/user = src.interactor
      src.fillReserves(user)
      ```
-     - Good:
+    * Good:
      ```
      var/user = interactor
      fillReserves(user)
      ```
```

Comment:
Perhaps add a section on use of the `p_their` system, as well as `UID()` over `\ref`

Reply:
addressed